cmd/osv-scanner/main_test.go

```diff
@@ -1520,6 +1520,48 @@ Filtered 2 vulnerabilities from output
 	}
 }
 
+// Tests all subcommands here.
+func TestRun_SubCommands(t *testing.T) {
+	t.Parallel()
+	tests := []cliTestCase{
+		// with scan subcommand
+		{
+			name:         "with scan subcommand",
+			args:         []string{"", "scan", "./fixtures/locks-many/composer.lock"},
+			wantExitCode: 0,
+			// The warning message exists here as we have a folder called 'scan' under './cmd/osv-scanner'
+			wantStdout: `
+				Scanning dir ./fixtures/locks-many/composer.lock
+				Scanned <rootdir>/fixtures/locks-many/composer.lock file and found 1 package
+				No issues found
+			`,
+			wantStderr: "Warning: 'scan' exists as both a subcommand of OSV-Scanner and as a file in the filesystem. It operates as a command here. If you intend to scan the file, please specify a subcommand.",
+		},
+		// scan with a flag
+		{
+			name:         "scan with a flag",
+			args:         []string{"", "scan", "--recursive", "./fixtures/locks-one-with-nested"},
+			wantExitCode: 0,
+			wantStdout: `
+				Scanning dir ./fixtures/locks-one-with-nested
+				Scanned <rootdir>/fixtures/locks-one-with-nested/nested/composer.lock file and found 1 package
+				Scanned <rootdir>/fixtures/locks-one-with-nested/yarn.lock file and found 1 package
+				No issues found
+			`,
+			wantStderr: "Warning: 'scan' exists as both a subcommand of OSV-Scanner and as a file in the filesystem. It operates as a command here. If you intend to scan the file, please specify a subcommand.",
```

Comment:
Can we add some testcases for when this error isn't output?

Reply:
When we specify the 'scan' subcommand, the error always shows here as we do have a folder called `./scan`. So I added some tests for `insertDefaultCommand()` to test this type of case.